### PR TITLE
feat: Make the installer offline

### PR DESCRIPTION
**Pull request by cat-master21**
WIP, updates the bootloader and forces it be offline plus this removes about 0.25 GB as the default pool doesn't have GCC12 and GCC13.

### CODE REVIEW

**Comment by cat-master21** [comment]
So for some reason dkms and other packages weren't removed/added but the installer works on UEFI but not on BIOS due to the iso not including `grub-pc`.

**Comment by cat-master21** [comment]
So how Lubuntu does it with their Calamares installer is they boot with `grub-pc` then install grub-efi if needed.

**Comment by oklopfer** [comment]
> So how Lubuntu does it with their Calamares installer is they boot with `grub-pc` then install grub-efi if needed.

This won't work on UEFI systems, though. The only way we could get it to do both UEFI + BIOS is by providing syslinux + grub-efi. Grub-pc is declared incompatible with grub-efi by live build.

**Comment by cat-master21** [comment]
Yes, thats why Calamares installs `grub-efi-amd64` if UEFI is detected and removes grub-pc yet Grub-pc is booted by both BIOS and UEFI in a live system @oklopfer.

**Comment by oklopfer** [comment]
> Yes, thats why Calamares installs `grub-efi-amd64` if UEFI is detected and removes grub-pc yet Grub-pc is booted by both BIOS and UEFI in a live system @oklopfer.

You are forgetting arm64 ISO platforms. grub-pc does not work for them. they need syslinux at least.

**Comment by cat-master21** [comment]
Alright so I can boot the image, available at https://github.com/rhino-linux/os/actions/runs/5913927031 if you want to test  unfortantely `grub-pc-bin` most also be installed. @oklopfer what do you mean, Arm64 uses grub-efi.

**Comment by oklopfer** [comment]
hey you should reply on discord tbh

**Comment by oklopfer** [comment]
but also realized we can't do this because of how apt needs to check with repos, allow-releaseinfo-change is vital for the rolling changes

File: etc/config/hooks/live/002-install-pacstall.chroot [hunk 1]
```diff
@@ -31,19 +32,31 @@ firmwareType:
...
+        -    command: apt-get update
              timeout: 120
-        -    command: apt install -y --no-upgrade -o Acquire::gpgv::Options::=--ignore-time-conflict grub-pc
-             timeout: 300
-        -    command: apt install -y --no-upgrade -o Acquire::gpgv::Options::=--ignore-time-conflict shim-signed
-             timeout: 300'
     "*":
         -    command: sed -i 's/ cdrom/ [trusted=yes] cdrom/g' /etc/apt/sources.list
-        -    timeout: 10
-        -    command: apt-get update --allow-releaseinfo-change
```
Comment (oklopfer) [inline etc/config/hooks/live/002-install-pacstall.chroot:43]: this is vital for checking if there was a base change (e.g. lunar -> mantic)

**Review by oklopfer — CHANGES_REQUESTED** [review]

File: etc/config/hooks/live/002-install-pacstall.chroot (hunk 1 above)
Comment (cat-master21) [reply etc/config/hooks/live/002-install-pacstall.chroot:43]: well the sources are still devel, and the CD is always the same. The user can update after install also considering that there is no internet.

**Pull request closed without merging**